### PR TITLE
[Bugfix] Fix TieredSpilloverCache flaky tests

### PR DESCRIPTION
**Pull request by peteralfonsi**
### Description
Fixes flaky tests in TieredSpilloverCacheStatsIT.java (and potentially flaky tests in CacheStatsAPIIndicesRequestCacheIT.java). 

Previously index refreshes could happen mid-test, which invalidated the cache and caused the number of hits, misses, etc, to be incorrect. This caused flakiness, and I confirmed setting the refresh interval to a very small value like 50 ms caused more frequent flakiness. I set the index refresh interval to -1 to prevent this from happening, and confirmed there were no longer failures when running each test for 1k iterations.

To avoid future issues, also perform a force merge after index creation, so that merges cannot happen mid-test either.

### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/14169, https://github.com/opensearch-project/OpenSearch/issues/14281, and https://github.com/opensearch-project/OpenSearch/issues/14029. 

### Check List
- [x] Functionality includes testing.
~- [N/A] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.~
~- [N/A] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peteralfonsi** [comment]
Flaky tests: 
https://github.com/opensearch-project/OpenSearch/issues/5426
https://github.com/opensearch-project/OpenSearch/issues/12407

File: modules/cache-common/src/internalClusterTest/java/org/opensearch/cache/common/tier/TieredSpilloverCacheStatsIT.java [hunk 1]
```diff
@@ -498,4 +504,16 @@ private static ImmutableCacheStatsHolder getNodeCacheStatsResult(Client client,
         NodeCacheStats ncs = nodeStatsResponse.getNodes().get(0).getNodeCacheStats();
         return ncs.getStatsByCache(CacheType.INDICES_REQUEST_CACHE);
     }
+
+    // Check each stat separately for more transparency if there's a failure
```
Comment (andrross) [inline modules/cache-common/src/internalClusterTest/java/org/opensearch/cache/common/tier/TieredSpilloverCacheStatsIT.java:508]: The problem with this approach is that if a new field is added to ImmutableCacheStats then it is possible this test will miss it and that field won't factor into the equality check here. Would simply adding a good toString() implementation on ImmutableCacheStats solve this problem?

Comment (peteralfonsi) [reply modules/cache-common/src/internalClusterTest/java/org/opensearch/cache/common/tier/TieredSpilloverCacheStatsIT.java:508]: Yes, good idea. Will make that change

**Comment by peteralfonsi** [comment]
Flaky test: 
https://github.com/opensearch-project/OpenSearch/issues/12407

**Review by andrross — APPROVED** [review]

**Pull request merged**